DragonSlayer.cpp

```diff
@@ -1,27 +1,48 @@
 #include "DragonSlayer.h"
 #include "Dragon.h"
+#include "Character.h"
+#include "Utility.h"
 
-//DragonSlayer::DragonSlayer
+DragonSlayer::DragonSlayer (std::string name_, int hp, int armor) : Character ( hp, armor, 4 ), name( name_ )
+{
+    helpfulItems = makeHelpfulItems( rand() % 4 );
+    defensiveItems = makeDefensiveItems( rand() % 3 );
+}
+
+DragonSlayer::~DragonSlayer(){}
 
-//DragonSlayer::getName
+const std::string& DragonSlayer::getName()
+{
+    return name;
+}
 
 void DragonSlayer::attack(Character& other)
 {
     std::cout << name << " is attacking " << other.getName() << " !!" << std::endl;
     if( auto* dragon = dynamic_cast<Dragon*>(&other) )
     {
-        assert(false);
+        //assert(false);
         //DragonSlayers get a 10x boost when attacking dragons, from their attack item.
         //so they should USE their attack item before attacking the dragon... 
         //
+
+        useAttackItem(this, &dragonSword);
+        
```

Comment:
call the member function on your attack item.
remember, items are 1-time-use only, so you need to reset it after use.  
look in the Character class for how the other item types get used/reset.
- [ ] revise.